xknx/dpt/dpt_1byte_control.py

```diff
@@ -0,0 +1,259 @@
+   Dimming is started with -/+100% (0x1/0x9) and keeps dimming until a STOP diagram (0x0/0x8) is received.
+
+As the same payload in these cases in interpreted completely different it is reasonable to make separate DPT classes.
+"""
+from enum import Enum
+
+from xknx.exceptions import ConversionError
+
+from .dpt import DPTBase
+
+
+class DPTControl(DPTBase):
+    """Abstraction for KNX B1U3 values (DPT 3.007/3.008)."""
+
+    # APCI (application layer control information)
+    APCI_CONTROLMASK = 0x08
+    APCI_STEPCODEMASK = 0x07
+    APCI_MAX_VALUE = APCI_CONTROLMASK | APCI_STEPCODEMASK
+
+    value_type = "control"
+    unit = ""
+    payload_length = 1
+
+    @classmethod
+    def _encode(cls, control, step_code):
+        """Encode control-bit with step-code."""
+        value = 1 if control > 0 else 0
+        value = (value << 3) | (step_code & cls.APCI_STEPCODEMASK)
+        return value
+
+    @classmethod
+    def _decode(cls, value):
+        """Decode value into control-bit and step-code."""
+        control = 1 if (value & cls.APCI_CONTROLMASK) != 0 else 0
+        step_code = value & cls.APCI_STEPCODEMASK
+        return control, step_code
+
+    @classmethod
+    def _test_boundaries(cls, raw):
+        """Test if raw KNX data is within defined range for this object."""
+        if isinstance(raw, int):
+            return 0 <= raw <= cls.APCI_MAX_VALUE
+        return False
+
+    @classmethod
+    def _test_values(cls, control, step_code):
+        """Test if input values are valid."""
+        if isinstance(control, int) and isinstance(step_code, int):
+            if control in (0, 1) and 0 <= step_code <= cls.APCI_STEPCODEMASK:
+                return True
+        return False
+
+    @classmethod
+    def to_knx(cls, value, invert=False):
+        """Serialize to KNX/IP raw data."""
+        if not isinstance(value, dict):
+            raise ConversionError(
+                "Cant serialize %s; invalid value type" % cls.__name__, value=value
+            )
+
+        try:
+            control = value["control"]
+            step_code = value["step_code"]
+        except KeyError:
+            raise ConversionError(
+                "Cant serialize %s; invalid keys" % cls.__name__, value=value
+            )
+
+        if not cls._test_values(control, step_code):
+            raise ConversionError(
+                "Cant serialize %s; invalid values" % cls.__name__, value=value
+            )
+
+        if invert:
+            control = 0 if control > 0 else 1
+
+        return cls._encode(control, step_code)
+
+    @classmethod
+    def from_knx(cls, raw, invert=False):
+        """Parse/deserialize from KNX/IP raw data."""
+        if not cls._test_boundaries(raw):
+            raise ConversionError("Cant parse %s" % cls.__name__, raw=raw)
+
+        control, step_code = cls._decode(raw)
+
+        if invert:
+            control = 0 if control > 0 else 1
+
+        return {"control": control, "step_code": step_code}
```

Comment:
I guess you meant `control ^ invert`. Would be possible, but do you really consider this easier to read? I find it somewhat crypting whereas the explicit version is more obvious. Plus, it would cause a mess if invert is not 0 or 1, which, however, is safe to assume as long as a boolean is passed.